libyul/backends/evm/ConstantOptimiser.cpp

```diff
@@ -126,7 +126,6 @@ Representation const& RepresentationFinder::findRepresentation(u256 const& _valu
 		return m_cache.at(_value);
 
 	yulAssert(m_dialect.auxiliaryBuiltinHandles().not_);
-	yulAssert(m_dialect.auxiliaryBuiltinHandles().shl);
```

Comment:
I honestly don't see a problem of also assert the support for `shl` here, even if it's redundant.

Reply:
the problem is that `shl` does not always exist for all evm versions - we could of course assert that it exists under the condition that the evm version is recent enough, but then again this is precisely what is captured in the implementation by guarding it with an if statement :)